docs/recipes/add_dataset.md

```diff
@@ -0,0 +1,112 @@
+
+To parse in code: Use `json.loads()` for JSON output.
+
+#### Example Output (JSON)
+```json
+{
+  "splits": ["train", "validation"],
+  "subsets": [],
+  "text_field_candidates": ["text"],
+  "sample_row": {
+    "text": "One day, a little girl named Lily found a toy in her room..."
+  }
+}
+```
+
+### Edge Cases and Advanced Handling
+- **Local vs. Hugging Face Datasets:** For local paths (e.g., Parquet, CSV, or directories), pass the path directly. The tool uses `load_dataset` with appropriate builders (e.g., "parquet" for .parquet files). Convert non-standard formats (e.g., custom JSON) to supported types via libraries like `pandas` beforehand if needed.
+- **Multi-Modal Datasets:** If non-text fields (e.g., images or audio) are present, they are noted in features but ignored for text-focused pretraining. Focus on `text_field_candidates` and flag for filtering if irrelevant.
+- **Large or Streaming Datasets:** For massive datasets, enable streaming internally (via `streaming=True` in advanced calls) to avoid full loading. Example: Add `--streaming` flag if implemented in future updates.
+- **No Text Field Found:** Review `text_field_candidates` for alternatives. If none, prepare for transformations in subsequent steps (e.g., concatenating fields).
+- **Errors:** Common issues include authentication (for private datasets) or format mismatches. Retry with explicit builder (e.g., `load_dataset(path, 'csv')`).
+
+## Rules for Agents
+These rules provide explicit guidance for AI agents (e.g., Codex, Claude, Cursor) to ensure consistent, reliable execution:
+- **Rule 1:** Prioritize the "text" field if present; otherwise, select the top candidate from `text_field_candidates` based on string dtype and relevance.
+- **Rule 2:** Always output the schema as JSON for easy parsing and downstream tooling.
+- **Rule 3:** If no text field is found or candidates are ambiguous, flag for human review and suggest a transformation (e.g., map "content" to "text").
+- **Rule 4:** Use the sample row to verify data format; if it contains non-text (e.g., binary), recommend skipping or filtering.
+- **Rule 5:** For datasets with subsets, evaluate each independently and merge schemas if they share structures.
+
+## Example Agent Prompt
+To test with an AI agent:
+> "Follow Step 1 of the add_dataset.md recipe. Inspect the schema for 'roneneldan/TinyStories' using the get_hf_dataset_schema tool. Apply all rules and output the full schema as JSON."
+
+## Testing and Validation
+- **Human Testing:** Run the examples above and verify the output matches your expectations. Cross-check with Hugging Face's dataset viewer for accuracy.
+- **Agent Testing:** Input this recipe into tools like Claude or Cursor, using the example prompt. Report issues or successes in the Marin Discord (#coding-roombas) or GitHub issues.
+- **Integration Testing:** Ensure the schema output integrates with Marin's pipelines (e.g., feed into token estimation tools in future steps).
```

Comment:
i'm a bit confused about these sections. Can we put fully organize this file into "guidelines for humans" "rules for agents"